content/5_Plat/HLD.mdx

```diff
@@ -15,6 +15,218 @@ frequency: 1
+- A  **light edge** connects the current node to any of its **light children**.
+- A  **heavy path** is the path formed by a collection **heavy edges**.
+- A  **light path** is the path formed by a collection **light edges**.
+
+**Properties:**
+
+Given the definitions stated above, an important property can be derived (I will prove this a bit later).
+
+- Any path from node $x$ to node $y$ on the tree can pass through at most $\mathcal{O}(\log N)$ light edges.
+
+Lets suppose the above property is true for a moment. Since a heavy path can only be broken by a light edge (or else the edge will be a part of the heavy path), we can know that there are at most $\mathcal{O}(\log N)$ heavy paths on any path from an arbitrary node $x$ to an arbitrary node $y$.
+
+In addition, by using segment trees (or any other RURQ data structure) we can calculate the value of any consecutive interval on any heavy path in $\mathcal{O}(\log N)$ time. 
+
+Since there are at most $\mathcal{O}(\log N)$ heavy paths and $\mathcal{O}(\log N)$ light edges, computing the value on the path from node $x$ to node $y$ will take $\mathcal{O}(\log^2 N + \log N)$ = $\mathcal{O}(\log^2 N)$ time. We can answer $Q$ queries in $\mathcal{O}(Q \log^2 N)$ time if the above property is true. This computational complexity allows us to pass questions where $N$ = $Q$ = $10^5$.
+
+**Proof:**
+
+Finally, we would need to prove the property above to conclude the tutorial. Consider a path from node $x$ to node $y$, this path can be broken down into a path from node $x$ to $LCA(x,y)$ and another path from node $y$ to $LCA(x,y)$. $LCA(x,y)$ must be an ancestor of both $x$ and $y$ by definition. Hence now we only need to prove that the path from any node to any of its ancestors can only pass through $\mathcal{O}(\log N)$ light edges to prove the property above. 
+
+Suppose that a light edge connects node $x$ and node $y$, where $y$ is a child of $x$. By the definition of light edge, $y$ must be a light child of $x$. This means that there has to be another child of $x$ with a larger subtree size than $y$, or else $y$ will become the heavy child of $x$. Hence the size of the subtree rooted at $x$ must be at least two times of the size of the subtree rooted at $y$. Since the subtree size of a node can not exceed the total number of nodes $N$, the doubling process can only happen at most $\mathcal{O}(\log N)$ times when traversing up from an arbitrary node to its parent along a light edge. Given this, we now know that there could be at most $\mathcal{O}(\log N)$ light edges in both the path from $x$ to $LCA(x,y)$ and the path from $LCA(x,y)$ to $y$. Hence there are at most $\mathcal{O}(\log N + \log N)$ = $\mathcal{O}(\log N)$ light edges in the path from an arbitrary node $x$ to an arbitrary node $y$. 
+
+We can solve the focus problem "Subtrees & Paths" using Heavy Light Decomposition:
+
+<LanguageSection>
+
+<CPPSection>
+
+```cpp
+
+#include <bits/stdc++.h>
```

Comment:
tabs

Reply:
Where specifically ? Indents in my code seems to be tabs on my side at least.

Reply:
most of your code uses 4 spaces to indent

![Screen Shot 2021-05-05 at 8 50 26 PM](https://user-images.githubusercontent.com/59348162/117239386-870eaf00-ade3-11eb-84b6-c377c85f9a9b.png)